thundra/plugins/trace/traceable.py

```diff
@@ -16,55 +21,69 @@ def __get_traceable_from_back_frame(frame):
             return _self
     return None
 
+
+def __get_funcname_from_back_frame(frame):
+    _back_frame = frame.f_back
+    if _back_frame and 'original_func' in _back_frame.f_locals:
+        _func = _back_frame.f_locals['original_func']
+        return _func.__name__
+    return None
+
+
+def __get_scope_from_back_frame(frame):
+    _back_frame = frame.f_back
+    if _back_frame and 'scope' in _back_frame.f_locals:
+        _scope = _back_frame.f_locals['scope']
+        if isinstance(_scope, Scope):
+            return _scope
+    return None
+
+
 def trace_lines(frame, event, arg):
     if event != 'line':
         return
 
-    _co = frame.f_code
-    _func_name = _co.co_name
     _line_no = frame.f_lineno
-    _filename = _co.co_filename
 
     _trace_local_variables_ = False
     _trace_lines_with_source = False
     _traceable = __get_traceable_from_back_frame(frame)
+    _scope = __get_scope_from_back_frame(frame)
+
+    if not _scope or not _scope.span:
+        return
+
     if _traceable:
         _trace_local_variables_ = _traceable._trace_local_variables
         _trace_lines_with_source = _traceable._trace_lines_with_source
 
-    _tracer = ThundraTracer.get_instance()
-
-    _active_scope = _tracer.scope_manager.active
-    _active_span = _active_scope.span if _active_scope is not None else None
-
-    if _active_span is not None and _active_span.class_name == 'Line':
-        _active_scope.close()
-
-    _scope = _tracer.start_active_span('@' + str(_line_no), finish_on_close=True)
-    _scope.span.class_name = 'Line'
-    _scope.span.on_started()
-
-    _line_source = ''
-    if _trace_lines_with_source:
-        _line_source = linecache.getline(_filename, _line_no).strip()
     _local_vars = []
     if _trace_local_variables_:
+        pickler = jsonpickle.pickler.Pickler(max_depth=3)
         for l in frame.f_locals:
             _local_var_value = frame.f_locals[l]
+            _local_var_type = type(_local_var_value).__name__
```

Comment:
Can this be problem if `_local_var_value` is `None`?

Reply:
No, type(None).__name__ is 'NoneType'